packages/headless/doc-parser/use-cases/commerce.ts

```diff
@@ -2,9 +2,133 @@ import {ActionLoaderConfiguration} from '../src/headless-export-resolvers/action
 import {ControllerConfiguration} from '../src/headless-export-resolvers/controller-resolver';
 import {EngineConfiguration} from '../src/headless-export-resolvers/engine-resolver';
 
-const controllers: ControllerConfiguration[] = [];
+const controllers: ControllerConfiguration[] = [
+  {
+    initializer: 'buildContext',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildCart',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildProductListing',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildProductListingBreadcrumbManager',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildProductListingCategoryFacet',
```

Comment:
We're introducing a concept of "sub-controllers" in the headless commerce sub-package that will render many "use case specific" controllers such as this one obsolete.

In fact, there's an open PR that actually removes those controllers altogether: https://github.com/coveo/ui-kit/pull/3857

The facet, pagination, and sort controllers builder functions will never be called directly by the implementer. What will happen will be more like this:

```typescript
const productListingController = buildProductListing(commerceEngine);

const facetGenerator = productListingController.facetGenerator();
for (const facet of facetGenerator.facets) {
  switch (facet.type) {
    case 'regular':
      // facet will be typed as RegularFacet, render it...
      break;
    case 'dateRange':
      // facet will be typed as DateFacet, render it...
      break;
    case 'numericalRange':
      // facet will be typed as NumericFacet, render it...
      break;
    case 'hierarchical':
      // facet will be typed as CategoryFacet, render it...
  }
}

// similar for sort controller (using productListingController.sort())
// and pagination controller (using productListingController.pagination())
```

The logic will be similar for the search controller (e.g., if you want to get the facet generator for a search controller you'd do `searchController.facetGenerator()`, and sort would be `searchController.sort()`, etc.). The recommendations controller also has a pagination sub-controller (but it doesn't have facets or sort).

Here's the implementation of the sub-controller controller: https://github.com/coveo/ui-kit/pull/3855/files#diff-70fb810df3b4d7646a54a4eb9c51a65a82fe48917db532f2018810b71ea340ecR74)